src/components/JobCard/JobCard.tsx

```diff
@@ -1,68 +1,77 @@
 import { Text, View, Pressable, Modal } from 'react-native';
 import styles from './CardStyles';
 import React, { useState } from 'react';
+import GestureRecognizer from 'react-native-swipe-gestures';
 
 interface JobCardProps {
-  title: string;
   description: string;
-  employer: string;
   hours: number;
-  salary: number;
+  employer: string;
   contact_info: string;
-  end_date: Date;
-  job_creator: string;
-  start_date: string;
+  id: string;
+  date: string;
+  companyName: string;
+  address: string;
+  contactPerson: string;
+  phone: string;
+  jobPosition: string;
+  languageRequirement: string;
+  workingHours: string;
+  workingDays: string;
+  salary: string;
+  probationPeriod: string;
+  employeeBenefit: string;
+  otherInfo: string;
+  hidden: Map<string, boolean>;
 }
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type, @typescript-eslint/naming-convention
-const JobCard = ({ title, description, employer, hours, salary, contact_info }: JobCardProps) => {
+const JobCard = ({ title, description, employer, hours, salary, contact_info, hidden }: JobCardProps) => {
   const [modalVisible, setModalVisible] = useState(false);
 
   return (
-    <View style={styles.cardContainer}>
-      <Modal
-        transparent={true}
-        visible={modalVisible}
-        animationType="slide"
-        onRequestClose={() => {
+    <Pressable
+      style={styles.cardContainer}
+      onPress={() => {
+        setModalVisible(true);
+      }}>
+      <GestureRecognizer
+        style={{ flex: 1 }}
+        onSwipeDown={() => {
           setModalVisible(!modalVisible);
         }}>
-        <View style={styles.centeredView}>
-          <View style={styles.modalView}>
-            <View style={styles.modalHeader}>
-              <Text style={styles.titleText}>{title}</Text>
-            </View>
-            <View style={styles.modalInfo}>
+        <Modal
+          transparent={true}
+          visible={modalVisible}
+          animationType="slide"
+          onRequestClose={() => {
+            setModalVisible(!modalVisible);
+          }}>
+          <View style={styles.centeredView}>
+            <View style={styles.modalView}>
+              <View style={styles.modalHeader}>
+                <Text style={styles.modalJobRefText}>{title}</Text>
```

Comment:
can we edit this so that this is the job id and not `title`? also, i don't think `title` is a field in the jobs object so I think this should be changed to `jobPosition`